### PR TITLE
feat: golang example simplify

### PR DESCRIPTION
**Pull request by Icemap**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [ ] master (the latest development version)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.2 (TiDB 7.2 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: develop/dev-guide-sample-application-golang-gorm.md [hunk 1]
```diff
@@ -1,198 +1,168 @@
 ---
-title: TiDB 和 GORM 的简单 CRUD 应用程序
-summary: 给出一个 TiDB 和 GORM 的简单 CRUD 应用程序示例。
+title: 使用 GORM 连接到 TiDB
+summary: 本文描述了 TiDB 和 GORM 的连接步骤，并给出了简单示例代码片段。
```
Comment (sykp241095) [inline develop/dev-guide-sample-application-golang-gorm.md:3]:
```suggestion
summary: 本文描述了 TiDB 和 GORM 的连接步骤，并给出了简单示例代码片段
```

Comment (sykp241095) [reply develop/dev-guide-sample-application-golang-gorm.md:3]: 水一个_

**Review by sykp241095 — APPROVED** [review]
LGTM

File: develop/dev-guide-sample-application-golang-sql-driver.md [hunk 2]
```diff
@@ -1,537 +1,269 @@
...
-    if len(players) != 0 {
-        if _, err := tx.Exec(buildBulkInsertSQL(len(players)), playerToArgs(players)...); err != nil {
-            tx.Rollback()
-            return err
-        }
-    }
+```golang
+openDB("mysql", func(db *sql.DB) {
+    insertSQL = "INSERT INTO player (id, coins, goods) VALUES (?, ?, ?)"
+    _, err := db.Exec(insertSQL, "id", 1, 1)
```
Comment (winkyao) [inline develop/dev-guide-sample-application-golang-sql-driver.md:183]: 第一列 是 "id"?, 这个 player 表好像没有看到建表？

id 还是用数字好一些？

Comment (Icemap) [reply develop/dev-guide-sample-application-golang-sql-driver.md:183]: 这个是 [sql-driver repository](https://github.com/tidb-samples/tidb-golang-sql-driver-quickstart/blob/main/README-zh.md) 的代码片段。为了尽量精简文档，建表过程都是没包含在文档内的。

File: develop/dev-guide-sample-application-golang-gorm.md (hunk 1 above)
Comment (Icemap) [reply develop/dev-guide-sample-application-golang-gorm.md:3]: 这样可能水不进去，可以提两个 commits，一个改过去，一个改回来 XD

File: develop/dev-guide-sample-application-golang-gorm.md [hunk 3]
```diff
@@ -1,198 +1,168 @@
...
+    ```
 
-```go
+6. 复制并粘贴对应连接字符串至 `.env` 中。需更改部分示例结果如下。
 
-package main
+    ```properties
+    TIDB_HOST='{gateway-region}.aws.tidbcloud.com'
+    TIDB_PORT='4000'
+    TIDB_USER='{prefix}.root'
```
Comment (sykp241095) [inline develop/dev-guide-sample-application-golang-gorm.md:71]:
```suggestion
    TIDB_USER='{user}'
```

File: develop/dev-guide-sample-application-golang-gorm.md [hunk 4]
```diff
@@ -1,198 +1,168 @@
...
-        bulkInsertPlayers[i] = Player{
-            ID:    uuid.New().String(),
-            Coins: rand.Intn(10000),
-            Goods: rand.Intn(10000),
-        }
-    }
+    ```properties
+    TIDB_HOST='{host}.clusters.tidb-cloud.com'
+    TIDB_PORT='4000'
+    TIDB_USER='{prefix}.root'
```
Comment (sykp241095) [inline develop/dev-guide-sample-application-golang-gorm.md:106]:
```suggestion
    TIDB_USER='{user}'
```

File: develop/dev-guide-sample-application-golang-sql-driver.md [hunk 5]
```diff
@@ -1,537 +1,269 @@
...
+5. 复制并粘贴对应的连接字符串至 `.env` 中。需更改部分示例结果如下。
 
-import (
-    "context"
-    "database/sql"
-)
+    ```properties
+    TIDB_HOST='{host}.clusters.tidb-cloud.com'
+    TIDB_PORT='4000'
+    TIDB_USER='{prefix}.root'
```
Comment (sykp241095) [inline develop/dev-guide-sample-application-golang-sql-driver.md:107]:
```suggestion
    TIDB_USER='{user}'
```

File: develop/dev-guide-sample-application-golang-sql-driver.md [hunk 6]
```diff
@@ -1,537 +1,269 @@
...
-    // Get a player.
-    testPlayer, err := getPlayer(db, "test")
-    if err != nil {
-        panic(err)
-    }
-    fmt.Printf("getPlayer: %+v\n", testPlayer)
+    ```properties
+    TIDB_HOST='{gateway-region}.aws.tidbcloud.com'
+    TIDB_PORT='4000'
+    TIDB_USER='{prefix}.root'
```
Comment (sykp241095) [inline develop/dev-guide-sample-application-golang-sql-driver.md:72]:
```suggestion
    TIDB_USER='{user}'
```

File: develop/dev-guide-sample-application-golang-sql-driver.md [hunk 7]
```diff
@@ -1,537 +1,269 @@
...
-```go
-package util
+5. 复制并粘贴对应的连接字符串至 `.env` 中。需更改部分示例结果如下。
 
-import (
-    "context"
-    "database/sql"
-)
+    ```properties
+    TIDB_HOST='{host}.clusters.tidb-cloud.com'
```
Comment (Mini256) [inline develop/dev-guide-sample-application-golang-sql-driver.md:105]: `{host}`

File: develop/dev-guide-sample-application-golang-gorm.md [hunk 8]
```diff
@@ -1,198 +1,179 @@
...
-    total, batch := 1919, 114
-    for i := 0; i < total; i++ {
-        bulkInsertPlayers[i] = Player{
-            ID:    uuid.New().String(),
-            Coins: rand.Intn(10000),
-            Goods: rand.Intn(10000),
-        }
-    }
+    ```dotenv
+    TIDB_HOST='{host}.clusters.tidb-cloud.com'
```
Comment (sykp241095) [inline develop/dev-guide-sample-application-golang-gorm.md:113]:
```suggestion
    TIDB_HOST='{host}'
```

File: develop/dev-guide-sample-application-golang-sql-driver.md [hunk 9]
```diff
@@ -1,537 +1,282 @@
...
+6. 复制并粘贴对应连接字符串至 `.env` 中。示例结果如下：
 
-    // Get a player.
-    testPlayer, err := getPlayer(db, "test")
-    if err != nil {
-        panic(err)
-    }
-    fmt.Printf("getPlayer: %+v\n", testPlayer)
+    ```dotenv
+    TIDB_HOST='{gateway-region}.aws.tidbcloud.com'
```
Comment (sykp241095) [inline develop/dev-guide-sample-application-golang-sql-driver.md:79]:
```suggestion
    TIDB_HOST='{host}'
```

as there is no concept of "region".

File: develop/dev-guide-sample-application-golang-sql-driver.md [hunk 10]
```diff
@@ -1,537 +1,282 @@
...
-```go
-package util
+5. 复制并粘贴对应的连接字符串至 `.env` 中。示例结果如下：
 
-import (
-    "context"
-    "database/sql"
-)
+    ```dotenv
+    TIDB_HOST='{host}.clusters.tidb-cloud.com'
```
Comment (sykp241095) [inline develop/dev-guide-sample-application-golang-sql-driver.md:114]:
```suggestion
    TIDB_HOST='{host}'
```

**Review by Oreoxmt — APPROVED** [review]

**Review by wd0517 — APPROVED** [review]

**Pull request merged**